pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,408 @@
+		type RoundIndex: Parameter
+			+ Member
+			+ MaybeSerializeDeserialize
+			+ sp_std::fmt::Debug
+			+ Default
+			+ sp_runtime::traits::MaybeDisplay
+			+ sp_runtime::traits::AtLeast32Bit
+			+ Copy;
+
+		/// Origin that is allowed to update the minimal orbiter deposit amount.
+		type UpdateMinOrbiterDepositOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn account_lookup_override)]
+	/// Account lookup override
+	pub type AccountLookupOverride<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, Option<T::AccountId>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn collators_pool)]
+	/// Current orbiters, with their "parent" collator
+	pub type CollatorsPool<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, CollatorPoolInfo<T::AccountId>>;
+
+	#[pallet::storage]
+	/// Current round index
+	pub(crate) type CurrentRound<T: Config> = StorageValue<_, T::RoundIndex, ValueQuery>;
+
+	#[pallet::storage]
+	/// Minimum deposit required to be registered as an orbiter
+	pub type MinOrbiterDeposit<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::storage]
+	/// Store active orbiter per round and per parent collator
+	pub(crate) type OrbiterPerRound<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		T::RoundIndex,
+		Blake2_128Concat,
+		T::AccountId,
+		T::AccountId,
+		OptionQuery,
+	>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			// Prune old OrbiterPerRound entries
+			let current_round = CurrentRound::<T>::get();
+			if current_round > T::MaxRoundArchive::get() {
+				let round_to_prune = current_round - T::MaxRoundArchive::get();
+				OrbiterPerRound::<T>::remove_prefix(round_to_prune, None);
+			}
+
+			0
+		}
+	}
+
+	/// An error that can occur while executing this pallet's extrinsics.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The collator is already added in orbiters program.
+		CollatorAlreadyAdded,
+		/// This collator is not in orbiters program.
+		CollatorNotFound,
+		/// There are already too many orbiters associated with this collator.
+		CollatorPoolTooLarge,
+		/// The minimum deposit required to register as an orbiter has not yet been included in the
+		/// onchain storage
+		MinOrbiterDepositNotSet,
+		/// This orbiter is already associated with this collator.
+		OrbiterAlreadyInPool,
+		/// Orbiter cant leave this round
+		OrbiterCantLeaveThisRound,
+		/// This orbiter has not made a deposit
+		OrbiterDepositNotFound,
+		/// This orbiter is not found
+		OrbiterNotFound,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Paid the orbiter account the balance as liquid rewards.
+		OrbiterRewarded {
+			account: T::AccountId,
+			rewards: BalanceOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add an orbiter in a collator pool
+		#[pallet::weight(500_000_000)]
```

Comment:
```suggestion
		#[pallet::weight(500_000_000)]
```
and benchmarking -> weights will also be done